src/Optimization/optimization_problem_results.jl

```diff
@@ -0,0 +1,829 @@
+"""
+function export_realized_results(res::Results)
+    save_path = mkpath(joinpath(get_output_dir(res), "export"))
+    return export_realized_results(res, save_path)
+end
+
+function export_realized_results(
+    res::Results,
+    save_path::AbstractString,
+)
+    if !isdir(save_path)
+        throw(IS.ConflictingInputsError("Specified path is not valid."))
+    end
+    write_data(read_results_with_keys(res, list_variable_keys(res)), save_path)
+    !isempty(list_dual_keys(res)) &&
+        write_data(
+            read_results_with_keys(res, list_dual_keys(res)),
+            save_path;
+            name = "dual",
+        )
+    !isempty(list_parameter_keys(res)) && write_data(
+        read_results_with_keys(res, list_parameter_keys(res)),
+        save_path;
+        name = "parameter",
+    )
+    !isempty(list_aux_variable_keys(res)) && write_data(
+        read_results_with_keys(res, list_aux_variable_keys(res)),
+        save_path;
+        name = "aux_variable",
+    )
+    !isempty(list_expression_keys(res)) && write_data(
+        read_results_with_keys(res, list_expression_keys(res)),
+        save_path;
+        name = "expression",
+    )
+    export_optimizer_stats(res, save_path)
+    files = readdir(save_path)
+    compute_file_hash(save_path, files)
+    @info("Files written to $save_path folder.")
+    return save_path
+end
+
+"""
+Save the optimizer statistics to CSV or JSON
+
+# Arguments
+
+  - `res::Union{OptimizationProblemResults, SimulationProblmeResults`: Results
+  - `directory::AbstractString` : target directory
+  - `format = "CSV"` : can be "csv" or "json
+"""
+function export_optimizer_stats(
+    res::Results,
+    directory::AbstractString;
+    format = "csv",
+)
+    data = read_optimizer_stats(res)
+    isnothing(data) && return
+    if uppercase(format) == "CSV"
+        CSV.write(joinpath(directory, "optimizer_stats.csv"), data)
+    elseif uppercase(format) == "JSON"
+        JSON.write(joinpath(directory, "optimizer_stats.json"), JSON.json(to_dict(data)))
+    else
+        throw(error("writing optimizer stats only supports csv or json formats"))
+    end
+end
+
+function write_data(
+    vars_results::Dict,
+    time::DataFrames.DataFrame,
+    save_path::AbstractString,
+)
+    for (k, v) in vars_results
+        var = DataFrames.DataFrame()
+        if size(time, 1) == size(v, 1)
+            var = hcat(time, v)
+        else
+            var = v
+        end
+        file_path = joinpath(save_path, "$(k).csv")
+        CSV.write(file_path, var)
+    end
+end
+
+function write_data(
+    data::DataFrames.DataFrame,
+    save_path::AbstractString,
+    file_name::String,
+)
+    if isfile(save_path)
+        save_path = dirname(save_path)
+    end
+    file_path = joinpath(save_path, "$(file_name).csv")
+    CSV.write(file_path, data)
+    return
+end
+
+# writing a dictionary of dataframes to files
+function write_data(vars_results::Dict, save_path::String; kwargs...)
```

Comment:
This is not a good name for a function in a generic package. `write_results`, `write_results_to_files`, `save_results`, `persist_results`?

Reply:
this would be another massive rename in PSI. I suggest we leave it and keep track of rename issues